test/getRecords.js

```diff
@@ -302,73 +302,75 @@ describe('getRecords', function() {
+            request(helpers.opts('GetShardIterator', {
+              StreamName: helpers.testStream,
+              ShardId: 'shardId-1',
+              ShardIteratorType: 'AT_TIMESTAMP',
+              Timestamp: secondInsertSeconds,
+            }), function(err, res) {
               if (err) return done(err)
               res.statusCode.should.equal(200)
 
-              delete res.body.MillisBehindLatest
+              var shardIterator = res.body.ShardIterator
 
-              helpers.assertShardIterator(res.body.NextShardIterator, helpers.testStream)
-              delete res.body.NextShardIterator
+              request(opts({ShardIterator: shardIterator}), function(err, res) {
+                if (err) return done(err)
+                res.statusCode.should.equal(200)
 
-              helpers.assertArrivalTimes(res.body.Records)
-              res.body.Records.forEach(function(record) { delete record.ApproximateArrivalTimestamp })
+                delete res.body.MillisBehindLatest
 
-              res.body.should.eql({
-                Records: [
-                  {
-                    PartitionKey: records2[1].PartitionKey,
-                    Data: records2[1].Data,
-                    SequenceNumber: recordsPut2[1].SequenceNumber,
-                  },
-                  {
-                    PartitionKey: records2[3].PartitionKey,
-                    Data: records2[3].Data,
-                    SequenceNumber: recordsPut2[3].SequenceNumber,
-                  },
-                  {
-                    PartitionKey: records2[5].PartitionKey,
-                    Data: records2[5].Data,
-                    SequenceNumber: recordsPut2[5].SequenceNumber,
-                  },
-                ],
-              })
+                helpers.assertShardIterator(res.body.NextShardIterator, helpers.testStream)
+                delete res.body.NextShardIterator
+
+                helpers.assertArrivalTimes(res.body.Records)
+                res.body.Records.forEach(function(record) { delete record.ApproximateArrivalTimestamp })
 
-              done()
+                res.body.should.eql({
+                  Records: [
+                    {
+                      PartitionKey: records2[1].PartitionKey,
+                      Data: records2[1].Data,
+                      SequenceNumber: recordsPut2[1].SequenceNumber,
+                    },
+                    {
+                      PartitionKey: records2[3].PartitionKey,
+                      Data: records2[3].Data,
+                      SequenceNumber: recordsPut2[3].SequenceNumber,
+                    },
+                    {
+                      PartitionKey: records2[5].PartitionKey,
+                      Data: records2[5].Data,
+                      SequenceNumber: recordsPut2[5].SequenceNumber,
+                    },
+                  ],
+                })
+
+                done()
+              })
             })
           })
-        })
+        }, 1)
```

Comment:
Can you explain this change? It seems like a hack of some sort

Reply:
@mhart Flow of this test looks like this:
request `PutRecords` with `records1` at `time1` 
request `PutRecords` with `records2` at `time2` (called `secondInsertSeconds`)
request `GetShardIterator` with `secondInsertSeconds` and type `AT_TIMESTAMP`

Sometimes, when `time1 == time2` this test fails as records from both inserts are returned. Example: https://travis-ci.org/mhart/kinesalite/builds/586195126

Thats why I have introduced minimal delay to make sure that `time1 != time2`